src/HealthChecks.EventStore/EventStoreHealthCheck.cs

```diff
@@ -10,45 +10,46 @@ public class EventStoreHealthCheck : IHealthCheck
         private const int ELAPSED_DELAY_MILLISECONDS = 500;
         private const int RECONNECTION_LIMIT = 1;
 
-        private readonly string _eventStoreConnection;
+        private readonly IEventStoreConnection? _eventStoreConnection;
+        private readonly string? _eventStoreConnectionString;
         private readonly string? _login;
         private readonly string? _password;
 
         public EventStoreHealthCheck(string eventStoreConnection, string? login, string? password)
         {
-            _eventStoreConnection = Guard.ThrowIfNull(eventStoreConnection);
+            _eventStoreConnectionString = Guard.ThrowIfNull(eventStoreConnection);
             _login = login;
             _password = password;
         }
 
+        public EventStoreHealthCheck(IEventStoreConnection eventStoreConnection)
+        {
+            _eventStoreConnection = eventStoreConnection ?? throw new ArgumentNullException(nameof(eventStoreConnection));
+        }
+
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                ConnectionSettingsBuilder connectionSettings;
+                var connectionSettingsBuilder = ConnectionSettings.Create()

```

Comment:
Let's move all code with `connectionSettingsBuilder` and `EventStoreConnection.Create` in method/local function and call it then like ` using (var connection = _eventStoreConnection ?? CreateEventStoreConnection())`.

Reply:
Done.